include/circt/Dialect/Arc/Ops.td

```diff
@@ -249,49 +249,81 @@ class MemoryAndDataTypesMatch<string mem, string data> : TypesMatchWith<
   "memory and data types must match", mem, data,
   "$_self.cast<MemoryType>().getWordType()">;
 
-def MemoryReadOp : ArcOp<"memory_read", [
+def MemoryReadPortOp : ArcOp<"memory_read_port", [
   MemoryEffects<[MemRead]>,
   MemoryAndDataTypesMatch<"memory", "data">
 ]> {
-  let summary = "Read word from memory";
+  let summary = "Read port from a memory";
   let arguments = (ins
     MemoryType:$memory,
     AnyInteger:$address,
-    I1:$clock,
+    Optional<I1>:$clock,
     I1:$enable
   );
   let results = (outs AnyInteger:$data);
   let assemblyFormat = [{
-    $memory `[` $address `]` `,` $clock `,` $enable
+    $memory `[` $address `]` `,` $enable ( `clock` $clock^ )?
     attr-dict `:` type($memory) `,` type($address)
```

Comment:
At a later point we might want to make `enable` optional as well. A lot of memory uses I've seen in the wild don't really provide an enable condition (it's just tied high). Purely cosmetic though, a later tweak down the road.

Reply:
Good point, I'll put it on my list!